backend/app/controllers/books_controller.rb

```diff
@@ -21,4 +33,8 @@ def create
   def book_params
     params.require(:book).permit(:title, :description)
   end
+
+  def set_book
+    @book = Book.find_by(id: params[:id])
```

Comment:
Memoization here as well. Also avoid using set_ for setters and get_ for getters just the name should be fine for clarity.